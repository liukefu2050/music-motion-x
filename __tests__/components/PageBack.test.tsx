import { fireEvent, render, screen } from "@testing-library/react"
import AppBack from "../../src/components/AppBack"
import { BrowserRouter } from "react-router-dom"

describe("AppBack", () => {
  it("worked with full props", async () => {
    const back = jest.fn()
    const originalBack = window.history.back
    Object.defineProperty(window.history, "back", {
      value: back,
      configurable: true,
      writable: true,
    })
    const { asFragment } = render(
      <BrowserRouter>
        <AppBack title="test" style={{ color: "red" }} isBlack />
      </BrowserRouter>,
    )
    fireEvent.click(screen.getByTestId("back-icon"))

    expect(back).toHaveBeenCalledTimes(1)
    Object.defineProperty(window.history, "back", {
      value: originalBack,
      configurable: true,
      writable: true,
    })

    expect(asFragment()).toMatchSnapshot()
  })
  it("worked with none black style", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <AppBack title="test" isBlack={false} />
      </BrowserRouter>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
