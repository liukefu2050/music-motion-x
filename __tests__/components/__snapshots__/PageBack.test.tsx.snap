// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBack worked with full props 1`] = `
<DocumentFragment>
  <div
    class="flex bg-mg pt-7 px-4 items-center"
    style="color: red;"
  >
    <img
      alt=""
      class=" w-4 h-4 mr-5"
      data-testid="back-icon"
      src="test-file-stub"
    />
    <p
      class="single_line text-base font-bold flex-1 text-fg text-center -indent-5 "
    >
      test
    </p>
    <a
      href="/discover"
    >
      <div
        class=" rounded-[50%] w-4 h-4 flex items-center justify-center cursor-pointer ml-2"
        style="border: 2px solid white;"
      >
        <div
          class=" rounded-[50%] w-2 h-2"
          style="border: 4px solid white;"
        />
      </div>
    </a>
  </div>
</DocumentFragment>
`;

exports[`AppBack worked with none black style 1`] = `
<DocumentFragment>
  <div
    class="flex bg-mg pt-7 px-4 items-center"
  >
    <img
      alt=""
      class=" w-4 h-4 mr-5"
      data-testid="back-icon"
      src="test-file-stub"
    />
    <p
      class="single_line text-base font-bold flex-1 text-fg text-center -indent-5 "
    >
      test
    </p>
    <a
      href="/discover"
    >
      <div
        class=" rounded-[50%] w-4 h-4 flex items-center justify-center cursor-pointer ml-2"
        style="border: 2px solid white;"
      >
        <div
          class=" rounded-[50%] w-2 h-2"
          style="border: 4px solid white;"
        />
      </div>
    </a>
  </div>
</DocumentFragment>
`;
